File: src/main/java/com/example/objects/pages/LoginPage/LoginPage.java
package com.example.objects.pages.LoginPage;

import com.codeborne.selenide.Selenide;
import com.codeborne.selenide.SelenideElement;
import com.example.extensions.AccountExtension;
import com.example.objects.account.Account;
import com.example.objects.pages.AccountPage.AccountPage;
import io.qameta.allure.Step;

import static com.codeborne.selenide.Selenide.*;
import static com.codeborne.selenide.Selectors.*;
import static com.codeborne.selenide.Condition.*;


public class LoginPage {

    public static final String url = "/accounts.gepur.com/signin";


    public SelenideElement emailField = $("#identifierId"),
                            nextButton = $("#identifierNext"),
                            passwordField = $(byName("password")),
                            loginButton = $("#passwordNext"),
                            errorMessage = $("div[aria-live='assertive']");


    @Step
    public LoginPage open(){
        Selenide.open(url);
        return this;
    }

    @Step("Login with account: {user.name} / {user.email}")
    public AccountPage loginAs(Account account){
        emailField.shouldBe(visible).setValue(account.getEmail());
        nextButton.click();
        passwordField.shouldBe(visible).setValue(account.getPassword());
        loginButton.click();
        return new AccountPage();
    }


    @Step("Login with account: {user.name} {user.email}")
    public AccountPage loginWith(AccountExtension admin){
        emailField.shouldBe(visible).setValue(admin.email).pressEnter();
        passwordField.shouldBe(visible).setValue(admin.password).pressEnter();
        return new AccountPage();
    }


    @Step
    public LoginPage setEmail(String email){
        emailField.shouldBe(visible).val(email);
        return this;
    }

    @Step
    public LoginPage setPassword(String email){
        emailField.shouldBe(visible).val(email);
        return this;
    }



}

File: src/main/kotlin/com/exmple/gepur/models/pages/LoginPage.kt
package com.exmple.gepur.models.pages

import com.exmple.gepur.extensions.kotlin_extensions.*
import com.exmple.gepur.extensions.kotlin_extensions.step

import com.exmple.gepur.models.stuff.User
import io.qameta.allure.Step


class LoginPage : WebPage(){


    override val url: String = "/auth/login"
    override val title: String = "Введите свой логин и пароль на Gepur"

    val emailField = s("#loginform-email")
    val passwordFiled = s("#loginform-password")


    @Step
    fun open(): LoginPage{
        return open(this.url)
    }

    @Step
    fun loginAs(user: User): MainPage {
        emailField.value = user.email
        passwordFiled.setValue(user.password).submit()
        return MainPage()
    }

    fun loginWith(user: User): LoginPage {
        return step("login with user: $user"){
            emailField.value = user.email
            passwordFiled.vaL(user.password).submit()
        }
    }





}
